Arrowgene.Ddon.Database/Sql/Core/DdonSqlDbCharacterPlayPointData.cs

```diff
@@ -0,0 +1,102 @@
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using System.Data.Common;
+
+namespace Arrowgene.Ddon.Database.Sql.Core
+{
+    public abstract partial class DdonSqlDb<TCon, TCom, TReader> : SqlDb<TCon, TCom, TReader>
+        where TCon : DbConnection
+        where TCom : DbCommand
+        where TReader : DbDataReader
+    {
+        protected static readonly string[] CDataJobPlayPointFields = new string[]
+        {
+            "character_id", "job", "play_point", "exp_mode"
+        };
+
+        private readonly string SqlInsertCharacterPlayPointData =
+            $"INSERT INTO \"ddon_character_playpoint_data\" ({BuildQueryField(CDataJobPlayPointFields)}) VALUES ({BuildQueryInsert(CDataJobPlayPointFields)});";
+
+        protected virtual string SqlInsertIfNotExistsCharacterPlayPointData { get; } =
+            $"INSERT INTO \"ddon_character_playpoint_data\" ({BuildQueryField(CDataJobPlayPointFields)}) SELECT {BuildQueryInsert(CDataJobPlayPointFields)} WHERE NOT EXISTS (SELECT 1 FROM \"ddon_character_playpoint_data\" WHERE \"character_id\" = @character_id AND \"job\" = @job);";
+
+        private static readonly string SqlUpdateCharacterPlayPointData =
+            $"UPDATE \"ddon_character_playpoint_data\" SET {BuildQueryUpdate(CDataJobPlayPointFields)} WHERE \"character_id\" = @character_id AND \"job\" = @job;";
+
+        private static readonly string SqlSelectCharacterPlayPointData =
+            $"SELECT {BuildQueryField(CDataJobPlayPointFields)} FROM \"ddon_character_playpoint_data\" WHERE \"character_id\" = @character_id AND \"job\" = @job;";
+
+        private static readonly string SqlSelectCharacterPlayPointDataByCharacter =
+            $"SELECT {BuildQueryField(CDataJobPlayPointFields)} FROM \"ddon_character_playpoint_data\" WHERE \"character_id\" = @character_id;";
+
+        private const string SqlDeleteCharacterPlayPointData = "DELETE FROM \"ddon_character_playpoint_data\" WHERE \"character_id\"=@character_id AND \"job\" = @job;";
+
+        public bool ReplaceCharacterPlayPointData(uint id, CDataJobPlayPoint replacedCharacterPlayPointData)
+        {
+            using TCon connection = OpenNewConnection();
+            return ReplaceCharacterPlayPointData(connection, id, replacedCharacterPlayPointData);
+        }
+
+        public bool ReplaceCharacterPlayPointData(TCon connection, uint id, CDataJobPlayPoint replacedCharacterPlayPointData)
```

Comment:
Why would we need to do this replace for play points? Shouldn't an update also work?